externals/CMakeLists.txt

```
@@ -138,6 +138,11 @@ endif()
 # MicroProfile
 add_library(microprofile INTERFACE)
 target_include_directories(microprofile SYSTEM INTERFACE ./microprofile)
+if (ENABLE_MICROPROFILE)
+    target_compile_definitions(microprofile INTERFACE MICROPROFILE_ENABLED=1)
+else()
+    target_compile_definitions(microprofile INTERFACE MICROPROFILE_ENABLED=0)
+endif()
```

Can't this be simplified to:
```suggestion
if (ENABLE_MICROPROFILE)
    target_compile_definitions(microprofile INTERFACE MICROPROFILE_ENABLED)
endif()
```
With `#ifdef`s used throughout?

---

microprofile is a external, so changing it to use ifdef is not a good idea. Better to keep it as original as possible. Simplifying this would require changing a lot of things and is not really needed.